Few-iOS/TableView.swift

```diff
@@ -104,19 +106,40 @@ public class TableView: Element {
     public override func applyDiff(old: Element, realizedSelf: RealizedElement?) {
         super.applyDiff(old, realizedSelf: realizedSelf)
         
-        if let scrollView = realizedSelf?.view as? FewTableView {
-            let handler = scrollView.handler
+        if let tableView = realizedSelf?.view as? FewTableView {
+            let handler = tableView.handler
             
             layoutElements()
             
             handler?.elements = elements
             
-            let tableView = scrollView.handler?.tableView
-            
+            let newHeaderView = header?.realize(realizedSelf).view
+            var needsUpdate = false
+            if newHeaderView != tableView.tableHeaderView {
+                needsUpdate = true
+            } else if let newHeight = newHeaderView?.frame.height where newHeight != tableView.rectForSection(0).minY {
+                needsUpdate = true
+            }
+            if needsUpdate {
+                tableView.tableHeaderView = newHeaderView
+                /// if you don't do this, the table view may compute the wrong header height
+                UIView.performWithoutAnimation {
+                    tableView.beginUpdates()
+                    tableView.endUpdates()
+                }
+            }
+
             handler?.selectionChanged = selectionChanged
         }
     }
     
+    public override func elementDidRealize(realizedSelf: RealizedElement) {
+        super.elementDidRealize(realizedSelf)
+        if let tableView = realizedSelf.view as! FewTableView? {
+            tableView.tableHeaderView = header?.realize(realizedSelf).view
```

Comment:
Is this the right place for this assignment? We need realized versions of both self and the header